src/clarity-angular/popover/signpost/signpost-trigger.directive.ts

```diff
@@ -28,13 +28,13 @@ export class SignpostTriggerDirective {
     constructor(private ifOpenService: IfOpenService) { }
 
     /**********
-     * @function onSignpostContentClick
+     * @function onSignpostTriggerClick
      *
      * @description
      * click handler for the Signpost trigger button used to hide/show SignpostContent.
      */
-    @HostListener("click")
+    @HostListener("click", ["$event"])
     onSignpostTriggerClick(): void {
-        this.ifOpenService.open = !this.ifOpenService.open;
+        this.ifOpenService.toggleWithEvent(event);
```

Comment:
This same code is repeated in dropdown and tooltip triggers, no? Hence the question I asked before, do you think a generic `clrTrigger` directive and generic `clrContent` and `clrMenu` components would suffice. Might make the API even simpler. trying to understand if this is possible and if not, why?